### ISSUE TITLE
[Bug] select with LIKE leads to BE crash

### ISSUE THREAD

**Issue by yoursweetie** [apache/doris#32848]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

branch-2.0 (doris-2.0.7-b1955a9)

### What's Wrong?

SELECT SQL with sub query and LIKE key words leads to BE crash

### What You Expected?

work

### How to Reproduce?

just execute sql as below and we'll find **all** BEs crash

`CREATE table tbl
(c1 varchar(31), c2 varchar(10))
ENGINE=OLAP
DUPLICATE KEY( c1 )
COMMENT "OLAP" 
DISTRIBUTED BY HASH( c1 ) BUCKETS  auto
PROPERTIES (
    "replication_num" = "1" 
);`

`insert into tbl select 'ab', 'abc';`

`SELECT c2 LIKE 'ab'
FROM (
    SELECT c2 
    FROM tbl
    EXCEPT 
    SELECT 'ab' c2
) AS t1;`

### Anything Else?

This resulted in all the BE nodes crashing, so I think it might be quite urgent.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by yoursweetie** [comment]
![堆栈](https://github.com/apache/doris/assets/52481734/250a89b1-efbc-4992-81ea-ee646cadbc6a)
![image](https://github.com/apache/doris/assets/52481734/13d8b9f2-a44b-4cdd-9881-122a8edc37cb)

**Comment by Mryange** [comment]
The subsequent version has been fixed.
```
mysql [test]>CREATE table tbl (c1 varchar(31), c2 varchar(10)) ENGINE=OLAP DUPLICATE KEY( c1 ) COMMENT "OLAP"  DISTRIBUTED BY HASH( c1 ) BUCKETS  auto PROPERTIES ( "replication_num" = "1"  );
ct 'ab', 'abc';
mysql [test]>insert into tbl select 'ab', 'abc';
mysql [test]>SELECT c2 LIKE 'ab' FROM ( SELECT c2  FROM tbl EXCEPT  SELECT 'ab' c2 ) AS t1;
+----------------+
| (c2 like 'ab') |
+----------------+
|              0 |
+----------------+
```

**Comment by yoursweetie** [comment]
> The subsequent version has been fixed.
> 
> ```
> mysql [test]>CREATE table tbl (c1 varchar(31), c2 varchar(10)) ENGINE=OLAP DUPLICATE KEY( c1 ) COMMENT "OLAP"  DISTRIBUTED BY HASH( c1 ) BUCKETS  auto PROPERTIES ( "replication_num" = "1"  );
> ct 'ab', 'abc';
> mysql [test]>insert into tbl select 'ab', 'abc';
> mysql [test]>SELECT c2 LIKE 'ab' FROM ( SELECT c2  FROM tbl EXCEPT  SELECT 'ab' c2 ) AS t1;
> +----------------+
> | (c2 like 'ab') |
> +----------------+
> |              0 |
> +----------------+
> ```

Can you tell me the exact version number, or which pull request fixed this issue? ty

**Comment by cambyzju** [comment]
> > The subsequent version has been fixed.
> > ```
> > mysql [test]>CREATE table tbl (c1 varchar(31), c2 varchar(10)) ENGINE=OLAP DUPLICATE KEY( c1 ) COMMENT "OLAP"  DISTRIBUTED BY HASH( c1 ) BUCKETS  auto PROPERTIES ( "replication_num" = "1"  );
> > ct 'ab', 'abc';
> > mysql [test]>insert into tbl select 'ab', 'abc';
> > mysql [test]>SELECT c2 LIKE 'ab' FROM ( SELECT c2  FROM tbl EXCEPT  SELECT 'ab' c2 ) AS t1;
> > +----------------+
> > | (c2 like 'ab') |
> > +----------------+
> > |              0 |
> > +----------------+
> > ```
> 
> I tested it on 3/26(yesterday) 2.0 branch, and I found it. Can you tell me the exact version number, or which pull request fixed this issue? ty

Do you have any non-default config in fe.conf, be.conf, or session variables?

**Comment by Mryange** [comment]
> > The subsequent version has been fixed.
> > ```
> > mysql [test]>CREATE table tbl (c1 varchar(31), c2 varchar(10)) ENGINE=OLAP DUPLICATE KEY( c1 ) COMMENT "OLAP"  DISTRIBUTED BY HASH( c1 ) BUCKETS  auto PROPERTIES ( "replication_num" = "1"  );
> > ct 'ab', 'abc';
> > mysql [test]>insert into tbl select 'ab', 'abc';
> > mysql [test]>SELECT c2 LIKE 'ab' FROM ( SELECT c2  FROM tbl EXCEPT  SELECT 'ab' c2 ) AS t1;
> > +----------------+
> > | (c2 like 'ab') |
> > +----------------+
> > |              0 |
> > +----------------+
> > ```
> 
> I tested it on 3/26(yesterday) 2.0 branch, and I found it. Can you tell me the exact version number, or which pull request fixed this issue? ty

This PR https://github.com/apache/doris/pull/26673 fixed the issue where the VEXCEPT operator would improperly execute the expr if open was not correct in the past. This PR includes relevant checks and fixes the errors. It has been merged into version 2.1 but has not yet been merged into version 2.0.
I'll try to pick that into 2.0 today.

**Comment by yoursweetie** [comment]
I tested it in 2.1 version yesterday, and this issue has been fixed, I want to figure out which PR fixed it with git bisect, but I found compile the old version requires old docker container, that's a big problem :monocle_face: . Any way, thank u! :tada:

**Comment by Mryange** [comment]
> I tested it in 2.1 version yesterday, and this issue has been fixed, I want to figure out which PR fixed it with git bisect, but I found compile the old version requires old docker container, that's a big problem 🧐 . Any way, thank u! 🎉
This PR https://github.com/apache/doris/pull/26673 fixed the issue where the VEXCEPT operator would improperly execute the expr if open was not correct in the past. This PR includes relevant checks and fixes the errors. It has been merged into version 2.1 but has not yet been merged into version 2.0.

**Comment by Mryange** [comment]
> I tested it in 2.1 version yesterday, and this issue has been fixed, I want to figure out which PR fixed it with git bisect, but I found compile the old version requires old docker container, that's a big problem 🧐 . Any way, thank u! 🎉


This PR https://github.com/apache/doris/pull/26673 fixed the issue where the VEXCEPT operator would improperly execute the expr if open was not correct in the past. This PR includes relevant checks and fixes the errors. It has been merged into version 2.1 but has not yet been merged into version 2.0.
